client/landing/stepper/declarative-flow/start-writing.ts

```diff
@@ -70,6 +70,14 @@ const startWriting: Flow = {
 		).getState();
 		const site = useSite();
 
+		// This flow clear the site_intent when flow is completed.
+		// We need to check if the site is launched and if so, clear the site_intent to avoid errors.
+		// See https://github.com/Automattic/dotcom-forge/issues/2886
+		const isSiteLaunched = site?.launch_status === 'launched' || false;
+		if ( isSiteLaunched ) {
+			setIntentOnSite( siteSlug, '' );
```

Comment:
What do you think about using `await` here? During my tests, at least twice I had to create and publish two new posts before the site_intent was removed. It's hard to reproduce, but it might be due to a race condition here.

Reply:
We can't use it here because `useStepNavigation` can't be an async function 😢 
I'm searching for an alternative

Reply:
I just updated using `useEffect`. This must solve the issue